DrugResistance/src/main/java/edu/stanford/hivdb/drugresistance/database/MutationComboScores.java

```diff
@@ -186,26 +186,26 @@ public MutationSet getRuleMutations() {
 
 				// TODO: can be optimized if mutation support "roughlyEquals",
 				// since actually this is just finding intersection items
-				MutationSet matchedMuts = cs.getRuleMutations()
-					.intersectsWith(geneSeqMuts);
 
-				if (matchedMuts.size() < cs.getRuleMutations().size()) {
-					// match all or fail
+				// match all or fail
+				if (!cs.getRuleMutations().equals(geneSeqMuts)) {
```

Comment:
This change breaks the algorithm since "match" doesn't indicate "equal":

For examples, Mutation sets `"RT:Q151M + RT:M184IV"` vs. `"RT:Q151M + RT:M184V"`, `"PR:V32I + PR:I84V"` vs. `""PR:V32VI + PR:I84VI" are both considered match, however they are obviously not equal.

Reply:
Thank you for clarifying. I'll change this line back and write some tests to capture this functionality. Also, it seems like accidentally left my comments on this code in a "Pending" state. So please ignore them you just received a notification.